tests/inputFields.spec.ts

```diff
@@ -0,0 +1,80 @@
+
+        const editPetTypeName = page.locator('#name')
+        await editPetTypeName.click()
+        await editPetTypeName.clear()
+        await editPetTypeName.fill('rabbit') //5. change pet name to rabbit and click update
+        const updatedPetTypeNameValue = await editPetTypeName.inputValue()
+        expect(updatedPetTypeNameValue).toEqual('rabbit')
+        const updateButton = page.getByRole('button', {name: "Update"})
+        await updateButton.click()
+
+        const petTypesListName = page.locator('tr input').nth(0)
+        await expect(petTypesListName).toHaveValue('rabbit') //6. assertion for first pet in list is rabbit
+
+        await page.getByRole('button', {name: "Edit"}).first().click() //7. click edit button for rabbit
+        await expect(page.getByRole('heading', {name: "Edit Pet Type"})).toBeVisible()
+        await editPetTypeName.click()
+        await editPetTypeName.clear()
+        await editPetTypeName.fill('cat') //8. change name to cat and update
+        await updateButton.click()
+        await expect(petTypesListName).toHaveValue('cat') //9. assertion that first pet in list is cat
+        
+    })
+
+    test('Cancel pet type update', async ({page}) => {
+        const petTypesTableName = page.getByRole('heading', {name: "Pet Types"})
+        await expect(petTypesTableName).toHaveText('Pet Types')  //2. assertion of Pet Types text displayed
+
+        await page.getByRole('button', {name: "Edit"}).nth(1).click() //3. click on edit button for dog
+        const editPetTypeName = page.locator('#name') //4. type new name moose
+        await editPetTypeName.click()
+        await editPetTypeName.clear()
+        await editPetTypeName.fill('moose')
+
+        const updatedPetTypeNameValue = await editPetTypeName.inputValue()
+        expect(updatedPetTypeNameValue).toEqual('moose') //5. Assertion for moose displayed on edit page
+
+        await page.getByRole('button', {name: "Cancel"}).click() //6. click cancel button
+
+        const petTypesListName = page.locator('tr input').nth(1)
+        await expect(petTypesListName).toHaveValue('dog') //7. assertion that dog is still displayed in the list of pet types
+    })
+
+    test('Pet type name validation', async ({page}) => {
+        const petTypesTableName = page.getByRole('heading', {name: "Pet Types"})
+        await expect(petTypesTableName).toHaveText('Pet Types')  //2. assertion of Pet Types text displayed
+
+        await page.getByRole('button', {name: "Edit"}).nth(2).click() //3. click on edit button for lizard
+        const editPetTypeName = page.locator('#name') //4. clear input field
+        await editPetTypeName.click()
+        await editPetTypeName.clear()
+
+        const nameIsRequiredMessage = page.locator('div .help-block').filter({hasText: "Name is required"}) //5. Assertion for 'name is required' message
+        expect(nameIsRequiredMessage).toBeVisible()
```

Comment:
This is also the same example as mentioned previously for the assertion of the text.
If the "Name is required" will not be displayed, the test will fail in line 79 with error "waiting for locator" rather than failing at assertion with a correct error message.
Update locator and assertion.

Reply:
Updated constant to `const nameIsRequiredMessage = page.locator('div .help-block')'
assertion to `expect(nameIsRequiredMessage).toHaveText('Name is required')`